src/Document/TextDocument.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Document;
+
+use Symfony\Component\Uid\Uuid;
+use Webmozart\Assert\Assert;
+
+final readonly class TextDocument
+{
+    public function __construct(
+        public Uuid $id,
+        public string $text,
```

Comment:
One last thing! I think I would rename it:
```suggestion
        public string $content,
```

It is the "content" of the text document .... 🤔